var/spack/repos/builtin/packages/cairo/autogen.patch

```diff
@@ -0,0 +1,16 @@
+--- a/autogen.sh	2021-05-27 13:57:07.000000000 -0400
++++ b/autogen.sh	2021-05-27 13:59:29.000000000 -0400
+@@ -13,13 +13,6 @@
+         exit 1
+ fi
+ 
+-GTKDOCIZE=`which gtkdocize`
+-if test -z $GTKDOCIZE; then
+-        echo "*** No GTK-Doc found, documentation won't be generated ***"
+-else
+-        gtkdocize || exit $?
+-fi
```

Comment:
I think I'm missing something: it looks like this handles the case where `gtkdocize` isn't found (i.e. in that case it doesn't run it). What error comes up without this patch?

Reply:
Oh yeah, sorry -- that discussion was briefly in a slack channel. Basically, the call to `autoreconf` will fail on `gtkdocize --copy` with `gtkdocize: command not found`. Since we're the only ones configuring, it doesn't matter if we don't copy documentation conf files; but if we don't have `gtkdocize`, even if we're building without docs, it'll fail without this patch or something similar.

Reply:
Apologies but I still don't understand the need for the changes in this file specifically - I think this explanation would be a good reason for the changes in `autoreconf/package.py`, but I still don't see why this code block has to be removed: it looks like if `which gtkdocize` fails that this outputs a message but otherwise succeeds, and if if it succeeds I don't think there would be a problem (since that means `gtkdocize` is available).

Is the existence of the `GTKDOCIZE` variable what leads to some later calling of `gtkdocize --copy` regardless of whether it is set or not?

Reply:
I completely misunderstood the context for your earlier comment (derp) so let me try this again.

Yes, the problem is with the GTKDOCIZE environment variable. It's originally set by the `autoreconf` change to the innocuous `echo`. However, the `autogen.sh` script clears the variable `GTKDOCIZE=` if `gtkdocize` is not found in the path. Subsequently the downstream `autoreconf` defaults `GTKDOCIZE=gtkdocize` because it was empty, and the autogen script fails.

So if the `autoregen.sh` used a different variable name it would be fine, but in its attempt to gracefully fail it ends  up undoing the `autoreconf` workaround we need to get autoconf 2.71 to pass.